source/extensions/http/injected_credentials/oauth2/oauth.h

```diff
@@ -18,11 +18,7 @@ namespace OAuth2 {
 class FilterCallbacks {
 public:
   virtual ~FilterCallbacks() = default;
-  enum class FailureReason {
-    StreamReset,
-    BadToken,
-    BadResponseCode,
-  };
+  enum class FailureReason { StreamReset, BadToken, BadResponseCode, BadTokenTransientFailure };
```

Comment:
Rather then `BadTokenTransientFailure`, please just give it a name of `EmptyResponse` or `EmptyToken`